src/Microsoft.DotNet.Interactive.SqlServer/MsSqlServiceClient.cs

```diff
@@ -158,10 +156,10 @@ public async Task<QueryExecuteSubsetResult> ExecuteQueryExecuteSubsetAsync(Query
 
         public async Task SendTextChangeNotificationAsync(Uri ownerUri, string newText, string oldText)
         {
-            var oldTextLines = oldText.Split('\n');
+            var oldTextLines = oldText.Split('\n').Select(text => text.EndsWith('\r') ? text[0..^1] : text).ToArray();
```

Comment:
You can simplify like this:

```csharp
text.Split(new []{ '\n', '\r' }, StringSplitOptions.RemoveEmptyEntries)
```

Reply:
Does that still preserve blank lines? We need the original number of lines in the text in order to get the right document change ranges.

Reply:
It does if you don't specify `StringSplitOptions.RemoveEmptyEntries`.

Reply:
Either way doesn't seem to work. When using this test string, "abc\r\ndef\r\n\r\nabc\r\nabcdef", the parsing returns 4 lines when specifying RemoveEmptyEntries, while it returns 9 when leaving out that option. We expect 5 lines given the Windows line endings in that string.

Reply:
Ah, righto.